src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.AndroidLibraries.targets

```diff
@@ -0,0 +1,115 @@
+@(AndroidLibrary) items and .aar files generation for class library
+projects.
+
+***********************************************************************************************
+-->
+<Project>
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.CreateAar"          AssemblyFile="$(_XamarinAndroidBuildTasksAssembly)" />
+  <UsingTask TaskName="Xamarin.Android.Tasks.ExtractJarsFromAar" AssemblyFile="$(_XamarinAndroidBuildTasksAssembly)" />
+
+  <ItemGroup>
+    <AvailableItemName Include="AndroidLibrary" />
+  </ItemGroup>
+  <ItemDefinitionGroup>
+    <AndroidLibrary>
+      <Bind>false</Bind>
+      <Pack>true</Pack>
+    </AndroidLibrary>
+  </ItemDefinitionGroup>
+  <PropertyGroup>
+    <_AarCacheFile>$(IntermediateOutputPath)$(TargetName).aar.cache</_AarCacheFile>
+    <_AarOutputPath>$(OutputPath)$(TargetName).aar</_AarOutputPath>
+  </PropertyGroup>
+
+  <Target Name="_CategorizeAndroidLibraries">
+    <ItemGroup Condition=" '$(AndroidApplication)' == 'true' ">
+      <AndroidAarLibrary    Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.aar' " />
+      <AndroidJavaLibrary   Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.jar' " />
+      <AndroidNativeLibrary Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.so' " />
+    </ItemGroup>
+    <ItemGroup Condition=" '$(AndroidApplication)' != 'true' ">
+      <AndroidAarLibrary     Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.aar' and '%(AndroidLibrary.Bind)' != 'true' " />
+      <LibraryProjectZip     Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.aar' and '%(AndroidLibrary.Bind)' == 'true' " />
+      <AndroidJavaLibrary    Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.jar' and '%(AndroidLibrary.Bind)' != 'true' " />
+      <EmbeddedJar           Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.jar' and '%(AndroidLibrary.Bind)' == 'true' " />
+      <EmbeddedNativeLibrary Include="@(AndroidLibrary)" Condition=" '%(AndroidLibrary.Extension)' == '.so' " />
+      <None Include="@(LibraryProjectZip)" CopyToOutputDirectory="PreserveNewest" Link="%(Filename)%(Extension)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ResolveAars" Condition=" '$(AndroidApplication)' == 'true' ">
+    <ItemGroup>
+      <_AarSearchDirectory   Include="@(_ReferencePath->'%(RootDir)%(Directory)')" />
+      <_AarSearchDirectory   Include="@(_ReferenceDependencyPaths->'%(RootDir)%(Directory)')" />
+      <_AarDistinctDirectory Include="@(_AarSearchDirectory->Distinct())" />
+      <!-- NOTE: %(AndroidSkipResourceProcessing) is required for located .aar's; there could be custom views -->
+      <AndroidAarLibrary     Update="@(AndroidAarLibrary)"                    AndroidSkipResourceProcessing="true" />
+      <AndroidAarLibrary     Include="%(_AarDistinctDirectory.Identity)*.aar" AndroidSkipResourceProcessing="false" />
```

Comment:
Is it possible we will end up with two of the same item here? One the user manually included via `AndroidAarLibrary ` and one we pick up from this `*.aar` ? What will happen to the metadata in that case? Do we need to take that into account?

Reply:
I'm still thinking about this one, let me test it.

Reply:
I tested putting the same path as this `_ResolveAars` target will find:
```
<AndroidAarLibrary Include="..\LibraryB\bin\Debug\net5.0-android\LibraryB.aar"  />
```
The build succeeds even with an `.aar` listed twice, but yes it would set `AndroidSkipResourceProcessing=false`:

![image](https://user-images.githubusercontent.com/840039/94956570-05078200-04b2-11eb-92f2-52f5cc62cca3.png)

The usage of the `.aar` seems OK? It seems to reuse the same `2` directory here:

![image](https://user-images.githubusercontent.com/840039/94956759-5152c200-04b2-11eb-8786-98e33fe8ad85.png)

Reply:
So does it then Convert the Resource cases for the duplicate aar? I'm thinking about targets that should not be run because we  did set `AndroidSkipResourceProcessing=True` but it was overridden..

Reply:
I updated the `DotNetBuildLibrary` test to check this now.

I think the new logic is better, as it will handle existing `.aar` files:

```xml
<_AarFromLibraries     Include="%(_AarDistinctDirectory.Identity)*.aar" />
<AndroidAarLibrary     Include="@(_AarFromLibraries)" Exclude="@(AndroidAarLibrary)" />
<!--
  NOTE: %(AndroidSkipResourceProcessing) is required for located .aar's; there could be custom views.
  Update in a second step, in case there is was an existing @(AndroidAarLibrary) with the same path.
-->
<AndroidAarLibrary Update="@(_AarFromLibraries)" AndroidSkipResourceProcessing="false" />
```

Seems to be working:

![image](https://user-images.githubusercontent.com/840039/95103785-ae40b900-06fa-11eb-91f5-6a8825fe5eac.png)